The job **`aarch64-gnu-llvm-21-1`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/25619575349/job/75203632511) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/75203632511) [(plain)](https://github.com/rust-lang/rust/commit/1ee8da5fd46bc6415c762cabdc09fc49b38e6f51/checks/75203632511/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
##[endgroup]
Executing "/scripts/stage_2_test_set1.sh"
+ /scripts/stage_2_test_set1.sh
+ '[' 1 == 1 ']'
+ echo 'PR_CI_JOB set; skipping tidy'
+ SKIP_TIDY='--skip tidy'
+ ../x.py --stage 2 test --skip tidy --skip compiler --skip src
PR_CI_JOB set; skipping tidy
##[group]Building bootstrap
    Finished `dev` profile [unoptimized] target(s) in 0.04s
##[endgroup]
---

5    |                                ^ unknown byte escape
6    |
7    = help: for more information, visit <https://doc.rust-lang.org/reference/tokens.html#literals>
+ help: if you meant to write a form feed, use a hex escape
+    |
+ LL - static FOO: &'static [u8] = b"\f";
+ LL + static FOO: &'static [u8] = b"\x0C";
+    |
8 
9 error: unknown byte escape: `f`
10   --> $DIR/byte-string-literals.rs:4:8

13    |        ^ unknown byte escape
14    |
15    = help: for more information, visit <https://doc.rust-lang.org/reference/tokens.html#literals>
+ help: if you meant to write a form feed, use a hex escape
+    |
+ LL -     b"\f";
+ LL +     b"\x0C";
+    |
16 
17 error: invalid character in numeric character escape: `Z`
18   --> $DIR/byte-string-literals.rs:5:10

Note: some mismatched output was normalized before being compared
- LL - static FOO: &'static [u8] = b"\f";  //~ ERROR unknown byte escape
- LL + static FOO: &'static [u8] = b"\\x0C";  //~ ERROR unknown byte escape
- LL -     b"\f";  //~ ERROR unknown byte escape
- LL +     b"\\x0C";  //~ ERROR unknown byte escape
+ help: if you meant to write a form feed, use a hex escape
+    |
+ LL - static FOO: &'static [u8] = b"\f";
+ LL + static FOO: &'static [u8] = b"\x0C";
+    |
+ help: if you meant to write a form feed, use a hex escape
+    |
+ LL -     b"\f";
+ LL +     b"\x0C";
+    |


The actual stderr differed from the expected stderr
To update references, rerun the tests and pass the `--bless` flag
To only update this specific test, also pass `--test-args parser/byte-string-literals.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/parser/byte-string-literals.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2" "--target=aarch64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/aarch64-unknown-linux-gnu/test/ui/parser/byte-string-literals" "-A" "unused" "-W" "unused_attributes" "-A" "internal_features" "-A" "incomplete_features" "-A" "unused_parens" "-A" "unused_braces" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/aarch64-unknown-linux-gnu/native/rust-test-helpers"
stdout: none
--- stderr -------------------------------
error: unknown byte escape: `f`
##[error]  --> /checkout/tests/ui/parser/byte-string-literals.rs:1:32
   |
LL | static FOO: &'static [u8] = b"\f";  //~ ERROR unknown byte escape
   |                                ^ unknown byte escape
   |
   = help: for more information, visit <https://doc.rust-lang.org/reference/tokens.html#literals>
help: if you meant to write a form feed, use a hex escape
   |
LL - static FOO: &'static [u8] = b"\f";  //~ ERROR unknown byte escape
LL + static FOO: &'static [u8] = b"\\x0C";  //~ ERROR unknown byte escape
   |

error: unknown byte escape: `f`
##[error]  --> /checkout/tests/ui/parser/byte-string-literals.rs:4:8
   |
LL |     b"\f";  //~ ERROR unknown byte escape
   |        ^ unknown byte escape
   |
   = help: for more information, visit <https://doc.rust-lang.org/reference/tokens.html#literals>
help: if you meant to write a form feed, use a hex escape
   |
LL -     b"\f";  //~ ERROR unknown byte escape
LL +     b"\\x0C";  //~ ERROR unknown byte escape
   |

error: invalid character in numeric character escape: `Z`
##[error]  --> /checkout/tests/ui/parser/byte-string-literals.rs:5:10
   |
LL |     b"\x0Z";  //~ ERROR invalid character in numeric character escape: `Z`
   |          ^ invalid character in numeric character escape

error: non-ASCII character in byte string literal
##[error]  --> /checkout/tests/ui/parser/byte-string-literals.rs:6:7
   |
LL |     b"é";  //~ ERROR non-ASCII character in byte string literal
   |       ^ must be ASCII
   |
help: if you meant to use the unicode code point for 'é', use a \xHH escape
   |
LL -     b"é";  //~ ERROR non-ASCII character in byte string literal
LL +     b"\xE9";  //~ ERROR non-ASCII character in byte string literal
   |

error: non-ASCII character in raw byte string literal
##[error]  --> /checkout/tests/ui/parser/byte-string-literals.rs:7:10
   |
LL |     br##"é"##;  //~ ERROR non-ASCII character in raw byte string literal
   |          ^ must be ASCII

error[E0766]: unterminated double quote byte string
##[error]  --> /checkout/tests/ui/parser/byte-string-literals.rs:8:6
   |
LL |       b"a  //~ ERROR unterminated double quote byte string
   |  ______^
LL | | }
   | |__^

error: aborting due to 6 previous errors
---

5    |                     ^ unknown byte escape
6    |
7    = help: for more information, visit <https://doc.rust-lang.org/reference/tokens.html#literals>
+ help: if you meant to write a form feed, use a hex escape
+    |
+ LL - static FOO: u8 = b'\f';
+ LL + static FOO: u8 = b'\x0C';
+    |
8 
9 error: unknown byte escape: `f`
10   --> $DIR/byte-literals.rs:6:8

13    |        ^ unknown byte escape
14    |
15    = help: for more information, visit <https://doc.rust-lang.org/reference/tokens.html#literals>
+ help: if you meant to write a form feed, use a hex escape
+    |
+ LL -     b'\f';
+ LL +     b'\x0C';
+    |
16 
17 error: invalid character in numeric character escape: `Z`
18   --> $DIR/byte-literals.rs:7:10

Note: some mismatched output was normalized before being compared
- LL - static FOO: u8 = b'\f';  //~ ERROR unknown byte escape
- LL + static FOO: u8 = b'\\x0C';  //~ ERROR unknown byte escape
- LL -     b'\f';  //~ ERROR unknown byte escape
- LL +     b'\\x0C';  //~ ERROR unknown byte escape
+ help: if you meant to write a form feed, use a hex escape
+    |
+ LL - static FOO: u8 = b'\f';
+ LL + static FOO: u8 = b'\x0C';
+    |
+ help: if you meant to write a form feed, use a hex escape
+    |
+ LL -     b'\f';
+ LL +     b'\x0C';
+    |


The actual stderr differed from the expected stderr
To update references, rerun the tests and pass the `--bless` flag
To only update this specific test, also pass `--test-args parser/byte-literals.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/parser/byte-literals.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2" "--target=aarch64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/aarch64-unknown-linux-gnu/test/ui/parser/byte-literals" "-A" "unused" "-W" "unused_attributes" "-A" "internal_features" "-A" "incomplete_features" "-A" "unused_parens" "-A" "unused_braces" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/aarch64-unknown-linux-gnu/native/rust-test-helpers"
stdout: none
--- stderr -------------------------------
error: unknown byte escape: `f`
##[error]  --> /checkout/tests/ui/parser/byte-literals.rs:3:21
   |
LL | static FOO: u8 = b'\f';  //~ ERROR unknown byte escape
   |                     ^ unknown byte escape
   |
   = help: for more information, visit <https://doc.rust-lang.org/reference/tokens.html#literals>
help: if you meant to write a form feed, use a hex escape
   |
LL - static FOO: u8 = b'\f';  //~ ERROR unknown byte escape
LL + static FOO: u8 = b'\\x0C';  //~ ERROR unknown byte escape
   |

error: unknown byte escape: `f`
##[error]  --> /checkout/tests/ui/parser/byte-literals.rs:6:8
   |
LL |     b'\f';  //~ ERROR unknown byte escape
   |        ^ unknown byte escape
   |
   = help: for more information, visit <https://doc.rust-lang.org/reference/tokens.html#literals>
help: if you meant to write a form feed, use a hex escape
   |
LL -     b'\f';  //~ ERROR unknown byte escape
LL +     b'\\x0C';  //~ ERROR unknown byte escape
   |

error: invalid character in numeric character escape: `Z`
##[error]  --> /checkout/tests/ui/parser/byte-literals.rs:7:10
   |
LL |     b'\x0Z';  //~ ERROR invalid character in numeric character escape: `Z`
   |          ^ invalid character in numeric character escape

error: byte constant must be escaped: `\t`
##[error]  --> /checkout/tests/ui/parser/byte-literals.rs:8:7
   |
LL |     b'    ';  //~ ERROR byte constant must be escaped
   |       ^^^^
   |
help: escape the character
   |
LL |     b'\t';  //~ ERROR byte constant must be escaped
   |       ++

error: byte constant must be escaped: `'`
##[error]  --> /checkout/tests/ui/parser/byte-literals.rs:9:7
   |
LL |     b''';  //~ ERROR byte constant must be escaped
   |       ^
   |
help: escape the character
   |
LL |     b'\'';  //~ ERROR byte constant must be escaped
   |       +

error: non-ASCII character in byte literal
##[error]  --> /checkout/tests/ui/parser/byte-literals.rs:10:7
   |
LL |     b'é';  //~ ERROR non-ASCII character in byte literal
   |       ^ must be ASCII
   |
help: if you meant to use the unicode code point for 'é', use a \xHH escape
   |
LL -     b'é';  //~ ERROR non-ASCII character in byte literal
LL +     b'\xE9';  //~ ERROR non-ASCII character in byte literal
   |

error[E0763]: unterminated byte constant
##[error]  --> /checkout/tests/ui/parser/byte-literals.rs:11:6
   |
LL |     b'a  //~ ERROR unterminated byte constant [E0763]
   |      ^^^^

error: aborting due to 7 previous errors

For more information about this error, try `rustc --explain E0763`.
```

</details>

--- a/compiler/rustc_parse/src/lexer/unescape_error_reporting.rs
+++ b/compiler/rustc_parse/src/lexer/unescape_error_reporting.rs
@@ -3,7 +3,7 @@
 use std::iter::once;
 use std::ops::Range;
 
-use rustc_errors::{Applicability, DiagCtxtHandle, ErrorGuaranteed};
+use rustc_errors::{Applicability, Diag, DiagCtxtHandle, ErrorGuaranteed};
 use rustc_literal_escaper::{EscapeError, Mode};
 use rustc_span::{BytePos, Span};
 use tracing::debug;
@@ -172,6 +172,8 @@ pub(crate) fn emit_unescape_error(
                     "for more information, visit \
                      <https://doc.rust-lang.org/reference/tokens.html#literals>",
                 );
+
+                foreign_escape_suggestion(&mut diag, (&ec, span), err_span);
             }
             diag.emit()
         }
@@ -300,6 +302,44 @@ pub(crate) fn emit_unescape_error(
             return None;
         }
     })
+}
+
+/// Add additional suggestions for escapes that are supported by C.
+fn foreign_escape_suggestion(
+    diag: &mut Diag<'_>,
+    (escaped_char, escape_span): (&str, Span),
+    err_span: Span,
+) {
+    if escaped_char == "?" {
+        diag.span_suggestion(
+            err_span,
+            "if you meant to write a literal question mark, don't escape the character",
+            "?",
+            Applicability::MaybeIncorrect,
+        );
+        return;
+    }
+
+    if u8::from_str_radix(escaped_char, 8).is_ok() {
+        diag.help(r"if you meant to write an ASCII control code, use a `\xNN` hex escape");
+        return;
+    }
+
+    let (name, hex) = match escaped_char {
+        "a" => ("an audible bell", "07"),
+        "b" => ("a backspace", "08"),
+        "f" => ("a form feed", "0C"),
+        "v" => ("a vertical tab", "0B"),
+        "e" => ("an ANSI escape sequence", "1B"),
+        _ => return,
+    };
+
+    diag.span_suggestion(
+        escape_span,
+        format!("if you meant to write {name}, use a hex escape"),
+        format!("x{hex}"),
+        Applicability::MaybeIncorrect,
+    );
 }
 
 /// Pushes a character to a message string for error reporting

--- a/tests/ui/parser/byte-literals.stderr
+++ b/tests/ui/parser/byte-literals.stderr
@@ -5,6 +5,11 @@ LL | static FOO: u8 = b'\f';
    |                     ^ unknown byte escape
    |
    = help: for more information, visit <https://doc.rust-lang.org/reference/tokens.html#literals>
+help: if you meant to write a form feed, use a hex escape
+   |
+LL - static FOO: u8 = b'\f';
+LL + static FOO: u8 = b'\x0C';
+   |
 
 error: unknown byte escape: `f`
   --> $DIR/byte-literals.rs:6:8
@@ -13,6 +18,11 @@ LL |     b'\f';
    |        ^ unknown byte escape
    |
    = help: for more information, visit <https://doc.rust-lang.org/reference/tokens.html#literals>
+help: if you meant to write a form feed, use a hex escape
+   |
+LL -     b'\f';
+LL +     b'\x0C';
+   |
 
 error: invalid character in numeric character escape: `Z`
   --> $DIR/byte-literals.rs:7:10

--- a/tests/ui/parser/byte-string-literals.stderr
+++ b/tests/ui/parser/byte-string-literals.stderr
@@ -5,6 +5,11 @@ LL | static FOO: &'static [u8] = b"\f";
    |                                ^ unknown byte escape
    |
    = help: for more information, visit <https://doc.rust-lang.org/reference/tokens.html#literals>
+help: if you meant to write a form feed, use a hex escape
+   |
+LL - static FOO: &'static [u8] = b"\f";
+LL + static FOO: &'static [u8] = b"\x0C";
+   |
 
 error: unknown byte escape: `f`
   --> $DIR/byte-string-literals.rs:4:8
@@ -13,6 +18,11 @@ LL |     b"\f";
    |        ^ unknown byte escape
    |
    = help: for more information, visit <https://doc.rust-lang.org/reference/tokens.html#literals>
+help: if you meant to write a form feed, use a hex escape
+   |
+LL -     b"\f";
+LL +     b"\x0C";
+   |
 
 error: invalid character in numeric character escape: `Z`
   --> $DIR/byte-string-literals.rs:5:10

--- a/tests/ui/parser/foreign-escapes.rs
+++ b/tests/ui/parser/foreign-escapes.rs
@@ -1,0 +1,25 @@
+// Specified by both C and Rust
+pub const SINGLE_QUOTE: char = '\'';
+pub const DOUBLE_QUOTE: char = '\"';
+pub const BACKSLASH: char = '\\';
+pub const NEWLINE: char = '\n';
+pub const CARRIAGE_RETURN: char = '\r';
+pub const HORIZONTAL_TAB: char = '\t';
+pub const NULL: char = '\0';
+
+// Specified by C, but not Rust
+pub const QUESTION_MARK: char = '\?'; //~ ERROR unknown character escape
+pub const AUDIBLE_BELL: char = '\a'; //~ ERROR unknown character escape
+pub const BACKSPACE: char = '\b'; //~ ERROR unknown character escape
+pub const FORM_FEED: char = '\f'; //~ ERROR unknown character escape
+pub const VERTICAL_TAB: char = '\v'; //~ ERROR unknown character escape
+pub const OCTAL: char = '\1'; //~ ERROR unknown character escape
+pub const OCTAL_TWO_DIGIT: char = '\12'; //~ ERROR unknown character escape
+pub const OCTAL_THREE_DIGIT: char = '\12'; //~ ERROR unknown character escape
+pub const OCTAL_OUT_OF_RANGE: char = '\9'; //~ ERROR unknown character escape
+
+// Not specified by C, but recognized by GCC as an extension.
+// Used for ANSI escape sequences in terminal emulators.
+pub const ESCAPE: char = '\e'; //~ ERROR unknown character escape
+
+fn main() {}

--- a/tests/ui/parser/foreign-escapes.stderr
+++ b/tests/ui/parser/foreign-escapes.stderr
@@ -1,0 +1,165 @@
+error: unknown character escape: `?`
+  --> $DIR/foreign-escapes.rs:11:35
+   |
+LL | pub const QUESTION_MARK: char = '\?';
+   |                                   ^ unknown character escape
+   |
+   = help: for more information, visit <https://doc.rust-lang.org/reference/tokens.html#literals>
+help: if you meant to write a literal backslash (perhaps escaping in a regular expression), consider a raw string literal
+   |
+LL - pub const QUESTION_MARK: char = '\?';
+LL + pub const QUESTION_MARK: char = r"\?";
+   |
+help: if you meant to write a literal question mark, don't escape the character
+   |
+LL - pub const QUESTION_MARK: char = '\?';
+LL + pub const QUESTION_MARK: char = '?';
+   |
+
+error: unknown character escape: `a`
+  --> $DIR/foreign-escapes.rs:12:34
+   |
+LL | pub const AUDIBLE_BELL: char = '\a';
+   |                                  ^ unknown character escape
+   |
+   = help: for more information, visit <https://doc.rust-lang.org/reference/tokens.html#literals>
+help: if you meant to write a literal backslash (perhaps escaping in a regular expression), consider a raw string literal
+   |
+LL - pub const AUDIBLE_BELL: char = '\a';
+LL + pub const AUDIBLE_BELL: char = r"\a";
+   |
+help: if you meant to write an audible bell, use a hex escape
+   |
+LL - pub const AUDIBLE_BELL: char = '\a';
+LL + pub const AUDIBLE_BELL: char = '\x07';
+   |
+
+error: unknown character escape: `b`
+  --> $DIR/foreign-escapes.rs:13:31
+   |
+LL | pub const BACKSPACE: char = '\b';
+   |                               ^ unknown character escape
+   |
+   = help: for more information, visit <https://doc.rust-lang.org/reference/tokens.html#literals>
+help: if you meant to write a literal backslash (perhaps escaping in a regular expression), consider a raw string literal
+   |
+LL - pub const BACKSPACE: char = '\b';
+LL + pub const BACKSPACE: char = r"\b";
+   |
+help: if you meant to write a backspace, use a hex escape
+   |
+LL - pub const BACKSPACE: char = '\b';
+LL + pub const BACKSPACE: char = '\x08';
+   |
+
+error: unknown character escape: `f`
+  --> $DIR/foreign-escapes.rs:14:31
+   |
+LL | pub const FORM_FEED: char = '\f';
+   |                               ^ unknown character escape
+   |
+   = help: for more information, visit <https://doc.rust-lang.org/reference/tokens.html#literals>
+help: if you meant to write a literal backslash (perhaps escaping in a regular expression), consider a raw string literal
+   |
+LL - pub const FORM_FEED: char = '\f';
+LL + pub const FORM_FEED: char = r"\f";
+   |
+help: if you meant to write a form feed, use a hex escape
+   |
+LL - pub const FORM_FEED: char = '\f';
+LL + pub const FORM_FEED: char = '\x0C';
+   |
+
+error: unknown character escape: `v`
+  --> $DIR/foreign-escapes.rs:15:34
+   |
+LL | pub const VERTICAL_TAB: char = '\v';
+   |                                  ^ unknown character escape
+   |
+   = help: for more information, visit <https://doc.rust-lang.org/reference/tokens.html#literals>
+help: if you meant to write a literal backslash (perhaps escaping in a regular expression), consider a raw string literal
+   |
+LL - pub const VERTICAL_TAB: char = '\v';
+LL + pub const VERTICAL_TAB: char = r"\v";
+   |
+help: if you meant to write a vertical tab, use a hex escape
+   |
+LL - pub const VERTICAL_TAB: char = '\v';
+LL + pub const VERTICAL_TAB: char = '\x0B';
+   |
+
+error: unknown character escape: `1`
+  --> $DIR/foreign-escapes.rs:16:27
+   |
+LL | pub const OCTAL: char = '\1';
+   |                           ^ unknown character escape
+   |
+   = help: for more information, visit <https://doc.rust-lang.org/reference/tokens.html#literals>
+   = help: if you meant to write an ASCII control code, use a `\xNN` hex escape
+help: if you meant to write a literal backslash (perhaps escaping in a regular expression), consider a raw string literal
+   |
+LL - pub const OCTAL: char = '\1';
+LL + pub const OCTAL: char = r"\1";
+   |
+
+error: unknown character escape: `1`
+  --> $DIR/foreign-escapes.rs:17:37
+   |
+LL | pub const OCTAL_TWO_DIGIT: char = '\12';
+   |                                     ^ unknown character escape
+   |
+   = help: for more information, visit <https://doc.rust-lang.org/reference/tokens.html#literals>
+   = help: if you meant to write an ASCII control code, use a `\xNN` hex escape
+help: if you meant to write a literal backslash (perhaps escaping in a regular expression), consider a raw string literal
+   |
+LL - pub const OCTAL_TWO_DIGIT: char = '\12';
+LL + pub const OCTAL_TWO_DIGIT: char = r"\12";
+   |
+
+error: unknown character escape: `1`
+  --> $DIR/foreign-escapes.rs:18:39
+   |
+LL | pub const OCTAL_THREE_DIGIT: char = '\12';
+   |                                       ^ unknown character escape
+   |
+   = help: for more information, visit <https://doc.rust-lang.org/reference/tokens.html#literals>
+   = help: if you meant to write an ASCII control code, use a `\xNN` hex escape
+help: if you meant to write a literal backslash (perhaps escaping in a regular expression), consider a raw string literal
+   |
+LL - pub const OCTAL_THREE_DIGIT: char = '\12';
+LL + pub const OCTAL_THREE_DIGIT: char = r"\12";
+   |
+
+error: unknown character escape: `9`
+  --> $DIR/foreign-escapes.rs:19:40
+   |
+LL | pub const OCTAL_OUT_OF_RANGE: char = '\9';
+   |                                        ^ unknown character escape
+   |
+   = help: for more information, visit <https://doc.rust-lang.org/reference/tokens.html#literals>
+help: if you meant to write a literal backslash (perhaps escaping in a regular expression), consider a raw string literal
+   |
+LL - pub const OCTAL_OUT_OF_RANGE: char = '\9';
+LL + pub const OCTAL_OUT_OF_RANGE: char = r"\9";
+   |
+
+error: unknown character escape: `e`
+  --> $DIR/foreign-escapes.rs:23:28
+   |
+LL | pub const ESCAPE: char = '\e';
+   |                            ^ unknown character escape
+   |
+   = help: for more information, visit <https://doc.rust-lang.org/reference/tokens.html#literals>
+help: if you meant to write a literal backslash (perhaps escaping in a regular expression), consider a raw string literal
+   |
+LL - pub const ESCAPE: char = '\e';
+LL + pub const ESCAPE: char = r"\e";
+   |
+help: if you meant to write an ANSI escape sequence, use a hex escape
+   |
+LL - pub const ESCAPE: char = '\e';
+LL + pub const ESCAPE: char = '\x1B';
+   |
+
+error: aborting due to 10 previous errors
+

